pallets/inactivity-tracking/src/lib.rs

```diff
@@ -418,11 +488,52 @@ pub mod pallet {
             );
             total_weight
         }
+
+        /// Internal function to mark a collator as offline.
+        pub fn mark_collator_offline(collator: &Collator<T>) -> DispatchResultWithPostInfo {
+            ensure!(
+                <EnableMarkingOffline<T>>::get(),
+                Error::<T>::MarkingOfflineNotEnabled
+            );
+            ensure!(
+                T::CollatorStakeHelper::is_candidate_selected(collator),
+                Error::<T>::CollatorNotEligibleCandidate
+            );
+            ensure!(
+                !<OfflineCollators<T>>::get(collator.clone()),
+                Error::<T>::CollatorNotOnline
+            );
+            ensure!(
+                !T::InvulnerablesFilter::is_invulnerable(collator),
+                Error::<T>::MarkingInvulnerableOfflineInvalid
+            );
+            <OfflineCollators<T>>::insert(collator.clone(), true);
+            T::CollatorStakeHelper::on_online_status_change(collator, false)?;
+            Self::deposit_event(Event::<T>::CollatorStatusUpdated {
+                collator: collator.clone(),
+                is_offline: true,
+            });
+            Ok(().into())
+        }
+
+        pub fn mark_collator_online(collator: &Collator<T>) -> DispatchResultWithPostInfo {
+            ensure!(
+                <OfflineCollators<T>>::get(collator),
+                Error::<T>::CollatorNotOffline
+            );
+            <OfflineCollators<T>>::insert(collator.clone(), false);
+            T::CollatorStakeHelper::on_online_status_change(collator, true)?;
```

Comment:
Add a comment here that this removes the elegible staking candidates list, and it needs to be called after updating the `OfflineCollators` map.

Reply:
Comments added in https://github.com/moondance-labs/tanssi/pull/881/commits/2e9c7eb6fa96ebe337f569d5e70853c0d53886c1